backend/lib/endpoints/posts.js

```diff
@@ -242,7 +242,7 @@ async function routes(app) {
         throw app.httpErrors.internalServerError();
       }
 
-      const { comments = [], numComments = 0 } = commentQuery;
+      const { comments = [], numComments = 0 } = commentQuery[0];
```

Comment:
I did approve but don't really understand the reason for this specific change. It's hard for me to confirm if this is right until all the v2 data model changes come in.

Reply:
The `groupBy` stage at the aggregation pipeline with a `null` group criteria will group all records into only one element (this way it also counts the comments).

So for this query, the results will always be in the first position as a single object, with the array of comments as an attribute, this way we would always be looking at the first element here.